### PR TITLE
深入 MNIST

### LINKED ISSUE #18: Deep MNIST for Experts
* Markdown文件：[文件地址](https://github.com/xitu/tensorflow/blob/zh-hans/tensorflow/docs_src/get_started/mnist/pros.md)
  * PR 地址：#39
  ----
  * 翻译时间：10 天
  * 校对时间：3 天
  * 翻译积分：10 分
  * 校对积分：3 分
  * [积分有什么用](https://github.com/xitu/gold-miner/wiki)

### PR DESCRIPTION
**Pull request by edvardHua**
翻译完啦。 resolve #18

### CODE REVIEW

**Comment by TobiasLee** [comment]
@leviding  校对认领

**Comment by caoyi0905** [comment]
@leviding  校对认领

**Comment by leviding** [comment]
@TobiasLee @caoyi0905 好的。

File: tensorflow/docs_src/get_started/mnist/pros.md [hunk 1]
```diff
@@ -321,15 +218,16 @@ We then convolve `x_image` with the weight tensor, add the
...
 h_conv1 = tf.nn.relu(conv2d(x_image, W_conv1) + b_conv1)
 h_pool1 = max_pool_2x2(h_conv1)
 ```
 
-### Second Convolutional Layer
+### 第二层卷积
 
-In order to build a deep network, we stack several layers of this type. The
-second layer will have 64 features for each 5x5 patch.
+为了构建一个更深的网络，我们会把几个类似的层堆叠起来。第二层中，每个5x5的patch会得到64个特征。
```
Comment (TobiasLee) [inline tensorflow/docs_src/get_started/mnist/pros.md]: 第二层中，每个5x5的patch会得到64个特征。 =>
第二层中，每个 5x5 的 patch 会得到 64 个特征。

File: tensorflow/docs_src/get_started/mnist/pros.md [hunk 2]
```diff
@@ -382,29 +270,21 @@ b_fc2 = bias_variable([10])
...
-How well does this model do? To train and evaluate it we will use code that is
-nearly identical to that for the simple one layer SoftMax network above.
+这个模型的效果如何呢？我们训练和评价模型的方式大体与前面评价单层 SoftMax 网络的差不多。
 
-The differences are that:
+区别主要有以下三点：
 
-- We will replace the steepest gradient descent optimizer with the more
-  sophisticated ADAM optimizer.
+- 我们性能更好的 ADAM 优化器，而不是梯度下降
```
Comment (TobiasLee) [inline tensorflow/docs_src/get_started/mnist/pros.md]: 我们性能更好的 ADAM 优化器，而不是梯度下降 =>
我们将使用性能更好的 ADAM 优化器，而不是梯度下降

File: tensorflow/docs_src/get_started/mnist/pros.md [hunk 3]
```diff
@@ -276,15 +186,9 @@ def bias_variable(shape):
...
+### 卷积和池化
 
-TensorFlow also gives us a lot of flexibility in convolution and pooling
-operations. How do we handle the boundaries? What is our stride size?
-In this example, we're always going to choose the vanilla version.
-Our convolutions uses a stride of one and are zero padded so that the
-output is the same size as the input. Our pooling is plain old max pooling
-over 2x2 blocks. To keep our code cleaner, let's also abstract those operations
-into functions.
+TensorFlow 在卷积和池化上有很强的灵活性。我们怎么处理边界？步长应该设多大？在这个实例里，我们会一直使用 vanilla 版本。我们的卷积使用 1 步长（stride size），0 边距（padding size）的模板，保证输出和输入是同一个大小。我们的池化用简单传统的 2x2 大小的模板做 max pooling。为了代码更简洁，我们把这部分抽象成一个函数。
```
Comment (caoyi0905) [inline tensorflow/docs_src/get_started/mnist/pros.md]: max pooling => 最大池

File: tensorflow/docs_src/get_started/mnist/pros.md [hunk 4]
```diff
@@ -295,23 +199,16 @@ def max_pool_2x2(x):
...
-### First Convolutional Layer
+### 第一层卷积
 
-We can now implement our first layer. It will consist of convolution, followed
-by max pooling. The convolution will compute 32 features for each 5x5 patch.
-Its weight tensor will have a shape of `[5, 5, 1, 32]`. The first two
-dimensions are the patch size, the next is the number of input channels, and
-the last is the number of output channels. We will also have a bias vector with
-a component for each output channel.
+现在我们可以开始实现第一层了。它由一个卷积接一个 max pooling 完成。卷积在每个 5x5 的 patch 中算出 32 个特征。卷积的权重张量形状是 [5, 5, 1, 32] ，前两个维度是 patch 的大小，接着是输入的通道数目，最后是输出的通道数目。 而对于每一个输出通道都有一个对应的偏置量。
```
Comment (caoyi0905) [inline tensorflow/docs_src/get_started/mnist/pros.md]: max pooling => 最大池

File: tensorflow/docs_src/get_started/mnist/pros.md [hunk 5]
```diff
@@ -1,270 +1,180 @@
...
 ```
 
-## Build a Multilayer Convolutional Network
+## 构建一个多层卷积网络
 
-Getting 92% accuracy on MNIST is bad. It's almost embarrassingly bad. In this
-section, we'll fix that, jumping from a very simple model to something
-moderately sophisticated: a small convolutional neural network. This will get us
-to around 99.2% accuracy -- not state of the art, but respectable.
+在 MNIST 上只有91%正确率，实在太糟糕。在这个小节里，我们用一个稍微复杂的模型：卷积神经网络来改善效果。这会达到大概99.2%的准确率。虽然不是最高，但是还是比较让人满意。
```
[the remark this answers is not in the document]
Comment (caoyi0905) [reply tensorflow/docs_src/get_started/mnist/pros.md]: 92% => 92 %

[the remark this answers is not in the document]
Comment (caoyi0905) [reply tensorflow/docs_src/get_started/mnist/pros.md]: 99.2% => 99.2 %

**Comment by caoyi0905** [comment]
@leviding 校对完成

**Comment by leviding** [comment]
@edvardHua 可以修改了

**Comment by leviding** [comment]
@edvardHua 抓紧修改了

File: tensorflow/docs_src/get_started/mnist/pros.md [hunk 6]
```diff
@@ -1,270 +1,180 @@
...
 
-### Start TensorFlow InteractiveSession
+### 运行 TensorFlow 的 InteractiveSession
 
-TensorFlow relies on a highly efficient C++ backend to do its computation. The
-connection to this backend is called a session.  The common usage for TensorFlow
-programs is to first create a graph and then launch it in a session.
+Tensorflow 依赖于一个高效的 C++ 后端来进行计算。与后端的这个连接叫做 session。一般而言，使用 TensorFlow 程序的流程是先创建一个图，然后在 session 中启动它。
+
+这里，我们使用更加方便的 `InteractiveSession` 类。通过它，你可以更加灵活地构建你的代码。它能让你在运行图的时候，插入一些计算图@{$get_started/get_started#the_computational_graph$computation graph}，这些计算图是由某些操作 (operations) 构成的。这对于工作在交互式环境中的人们来说非常便利，比如使用 IPython 。如果你没有使用 `InteractiveSession`，那么你需要在启动 session 之前构建整个计算图，然后启动该计算图@{$get_started/get_started#the_computational_graph$launching the graph}。
```
[the remark this answers is not in the document]
Comment (edvardHua) [reply tensorflow/docs_src/get_started/mnist/pros.md]: 棒棒哒

File: tensorflow/docs_src/get_started/mnist/pros.md [hunk 7]
```diff
@@ -1,270 +1,180 @@
...
+这里的 `x` 和 `y_` 并不是特定的值，相反，他们都只是一个占位符，可以在 TensorFlow 运行某一计算时根据该占位符输入具体的值。
 
-The input images `x` will consist of a 2d tensor of floating point numbers.
-Here we assign it a `shape` of `[None, 784]`, where `784` is the dimensionality
-of a single flattened 28 by 28 pixel MNIST image, and `None` indicates that the
-first dimension, corresponding to the batch size, can be of any size.  The
-target output classes `y_` will also consist of a 2d tensor, where each row is a
-one-hot 10-dimensional vector indicating which digit class (zero through nine)
-the corresponding MNIST image belongs to.
+输入图片 `x` 是一个 2 维的浮点数张量。这里，分配给它的 shape 为 [None, 784] ，其中 784 是一张展平的 MNIST 28 * 28 尺寸图片的维度。None 表示其值大小不定，在这里作为第一个维度值，用以指代 batch 的大小，意即 x 的数量不定。输出类别值 `y_` 也是一个2维张量，其中每一行为一个 10 维的 one-hot 向量,用于代表对应某一MNIST 图片的类别（从一到九）。
```
[the remark this answers is not in the document]
Comment (edvardHua) [reply tensorflow/docs_src/get_started/mnist/pros.md]: 但实际上，在设置了 batch 参数后，每次迭代的输入图片数量又变成确定的了。

**Comment by edvardHua** [comment]
@leviding 修改完了。

File: tensorflow/docs_src/get_started/mnist/pros.md [hunk 8]
```diff
@@ -1,270 +1,180 @@
...
 ```
 
-## Build a Multilayer Convolutional Network
+## 构建一个多层卷积网络
 
-Getting 92% accuracy on MNIST is bad. It's almost embarrassingly bad. In this
-section, we'll fix that, jumping from a very simple model to something
-moderately sophisticated: a small convolutional neural network. This will get us
-to around 99.2% accuracy -- not state of the art, but respectable.
+在 MNIST 上只有 92 % 正确率，实在太糟糕。在这个小节里，我们用一个稍微复杂的模型：卷积神经网络来改善效果。这会达到大概 99.2 % 的准确率。虽然不是最高，但是还是比较让人满意。
```
Comment (leviding) [inline tensorflow/docs_src/get_started/mnist/pros.md]: 数字和符号之间不要空格，全文检查下吧

File: tensorflow/docs_src/get_started/mnist/pros.md [hunk 9]
```diff
@@ -427,9 +307,8 @@ with tf.Session() as sess:
       x: mnist.test.images, y_: mnist.test.labels, keep_prob: 1.0}))
 ```
 
-The final test set accuracy after running this code should be approximately 99.2%.
+跑完测试集后精确度应该接近 99.2 %。
```
Comment (leviding) [inline tensorflow/docs_src/get_started/mnist/pros.md]: 同

Comment (caoyi0905) [reply tensorflow/docs_src/get_started/mnist/pros.md]: 啊。。这个是我是失误。。。

**Comment by edvardHua** [comment]
@leviding 好了。

**Comment by leviding** [comment]
@edvardHua Good job!
